### PR TITLE
Set up changesets and a basic release process

### PR DESCRIPTION
**Pull request by hsubox76**
Use [changesets](https://github.com/changesets/changesets/tree/main) for versioning.

Publish process is:
- On every push to `main`, a Version Packages PR is created or updated by changesets-action
- When we are ready to release, we merge the Version Packages PR to main, which triggers an NPM publish workflow

### CODE REVIEW

File: .github/workflows/create-release-pr.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Create Release PR
```
Comment (DellaBitta) [inline .github/workflows/create-release-pr.yml:1]: Add a copyright and license header.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Publish Release
```
Comment (DellaBitta) [inline .github/workflows/publish.yml:1]: Same here.

File: .github/workflows/create-release-pr.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  release:
+    name: Create Release PR
+    runs-on: ubuntu-latest
+    if: ${{ !startsWith(github.event.head_commit.message, 'Version Packages') }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 18
```
Comment (DellaBitta) [inline .github/workflows/create-release-pr.yml:19]: It says 18.x in the next workflow. Could you keep the names consistent? (either one is fine with me)

**Review by DellaBitta — APPROVED** [review]
approved with some notes.

File: .github/workflows/create-release-pr.yml (hunk 3 above)
Comment (hsubox76) [reply .github/workflows/create-release-pr.yml:33]: Set to 18.

**Review by lahirumaramba — APPROVED** [review]

**Pull request merged**